src/api/libcellml/model.h

```diff
@@ -67,6 +67,78 @@ class LIBCELLML_EXPORT Model: public Nameable
      */
     void addComponent(const Component &c);
 
+    /**
+     * @brief remove the component at the given index.
+     * Remove the component at the given index.  If the index is
+     * not in the range 0, componentCount then an exception will
+     * be raised.
+     * @param index the index of the component to remove.
+     */
+    void removeComponent(size_t index);
+
+    /**
+     * @brief remove the component at the given index.
+     * Remove the component with the given name.  This method will
+     * remove the first component that is found and return.  If the
+     * name is not found no component is removed.
+     * @param name the name of the component to remove.
+     */
+    void removeComponent(const std::string &name);
+
+    /**
+     * @brief tests to see if the component is contained within the Model.
+     * Tests to see if the component is contained within the Model.  Returns
+     * true if the component is in the Model and false otherwise.
+     * @param name the name of the component to test for existence in the Model.
+     * @return true if the named component is in the model and false otherwise.
+     */
+    bool containsComponent(const std::string &name);
+
+    /**
+     * @brief get a component at index.
+     * Returns a const reference to a component at the given index.  If the
+     * index is not valid std::runtime_error exception is thrown.
+     * @param index The index of the component to return
+     * @return a reference to the component at the given index
+     */
+    const Component& getComponent(size_t index) const;
+
+    /**
+     * @brief get a component at index.
+     * Returns a reference to a component at the given index.  If the index
+     * is not valid std::runtime_error exception is thrown.
+     * This is an overloaded method.
+     * @param index The index of the component to return
+     * @return a reference to the component at the given index
+     */
+    Component& getComponent(size_t index);
+
+    /**
+     * @brief take the component at the given index and return it.
+     * Removes the component at the given index position and returns it.
+     * If an invalid index is passed to the method a std::runtime_error
+     * exception is thrown.
+     * @param index the index of the component to take.
+     * @return the component at the given index.
+     */
+    Component takeComponentAt(size_t index);
```

Comment:
following the other `*Component` methods, I would expect to have a `takeComponent(size_t index)` and a `takeComponent(const std::string& name)` method.